config/org.eclipse.jdt.core.prefs

```diff
@@ -1,37 +1,60 @@
 org.eclipse.jdt.core.compiler.annotation.inheritNullAnnotations=enabled
+org.eclipse.jdt.core.compiler.problem.APILeak=error
+org.eclipse.jdt.core.compiler.problem.annotationSuperInterface=error
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
 org.eclipse.jdt.core.compiler.problem.comparingIdentical=error
 org.eclipse.jdt.core.compiler.problem.deadCode=error
+org.eclipse.jdt.core.compiler.problem.deadCodeInTrivialIfStatement=error
 org.eclipse.jdt.core.compiler.problem.deprecationInDeprecatedCode=enabled
 org.eclipse.jdt.core.compiler.problem.deprecationWhenOverridingDeprecatedMethod=enabled
 org.eclipse.jdt.core.compiler.problem.discouragedReference=error
+org.eclipse.jdt.core.compiler.problem.emptyStatement=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
 # The severity is set to 'warning', since we need to suppress it in some cases.
 org.eclipse.jdt.core.compiler.problem.explicitlyClosedAutoCloseable=warning
-org.eclipse.jdt.core.compiler.problem.emptyStatement=error
 org.eclipse.jdt.core.compiler.problem.fallthroughCase=error
-org.eclipse.jdt.core.compiler.problem.fatalOptionalError=disabled
 org.eclipse.jdt.core.compiler.problem.finallyBlockNotCompletingNormally=error
 org.eclipse.jdt.core.compiler.problem.finalParameterBound=error
+org.eclipse.jdt.core.compiler.problem.forbiddenReference=error
 org.eclipse.jdt.core.compiler.problem.hiddenCatchBlock=error
-org.eclipse.jdt.core.compiler.problem.includeNullInfoFromAsserts=enabled
 org.eclipse.jdt.core.compiler.problem.incompatibleNonInheritedInterfaceMethod=error
 org.eclipse.jdt.core.compiler.problem.incompleteEnumSwitch=error
 org.eclipse.jdt.core.compiler.problem.indirectStaticAccess=error
+org.eclipse.jdt.core.compiler.problem.invalidJavadoc=error
+org.eclipse.jdt.core.compiler.problem.invalidJavadocTags=error
+org.eclipse.jdt.core.compiler.problem.invalidJavadocTagsDeprecatedRef=error
+org.eclipse.jdt.core.compiler.problem.invalidJavadocTagsNotVisibleRef=error
+org.eclipse.jdt.core.compiler.problem.invalidJavadocTagsVisibility=error
 org.eclipse.jdt.core.compiler.problem.localVariableHiding=error
 org.eclipse.jdt.core.compiler.problem.methodWithConstructorName=error
 org.eclipse.jdt.core.compiler.problem.missingDefaultCase=error
+org.eclipse.jdt.core.compiler.problem.missingDeprecatedAnnotation=error
 org.eclipse.jdt.core.compiler.problem.missingEnumCaseDespiteDefault=enabled
 org.eclipse.jdt.core.compiler.problem.missingHashCodeMethod=error
+#We use checkstyle rule to ensure that all methods and types are documented.
+org.eclipse.jdt.core.compiler.problem.missingJavadocComments=ignore
```

Comment:
It shouldn't matter if we already have a check to ensure missing javadocs. As long it isn't causing a conflict with another rule it should be enabled. Since you didn't mention there is a conflict, I assume this isn't the case.
Also as I stated at https://github.com/checkstyle/checkstyle/pull/6331#pullrequestreview-188451359 new ones should be moved below https://github.com/checkstyle/checkstyle/blob/master/config/org.eclipse.jdt.core.prefs#L73 as they will be finished in another issue.

Reply:
Interesting that this check was long before R10.
I suppose we could add it. But we will need to make two different configurations. One for the main code and another for the tests. We usually don't have javadoc comments in tests.